x/dsmr/certificate.go

```diff
@@ -1,68 +1,48 @@
 // Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
+//go:generate go run github.com/StephenButtolph/canoto/canoto --concurrent=false $GOFILE
+
 package dsmr
 
 import (
 	"context"
-	"errors"
 	"fmt"
 
-	"github.com/ava-labs/avalanchego/codec/linearcodec"
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/snow/validators"
-	"github.com/ava-labs/avalanchego/utils/units"
-	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
 	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
-
-	"github.com/ava-labs/hypersdk/codec"
-
-	acodec "github.com/ava-labs/avalanchego/codec"
-)
-
-const (
-	CodecVersion = 0
-
-	MaxMessageSize = units.KiB
 )
 
-var Codec acodec.Manager
-
-func init() {
-	Codec = acodec.NewManager(MaxMessageSize)
-	lc := linearcodec.NewDefault()
+type ChunkReference struct {
+	ChunkID  ids.ID     `canoto:"fixed bytes,1"`
+	Producer ids.NodeID `canoto:"fixed bytes,2"`
+	Expiry   int64      `canoto:"int,3"`
 
-	err := errors.Join(
-		Codec.RegisterCodec(CodecVersion, lc),
-	)
-	if err != nil {
-		panic(err)
-	}
+	canotoData canotoData_ChunkReference
 }
 
-type ChunkReference struct {
-	ChunkID  ids.ID     `serialize:"true"`
-	Producer ids.NodeID `serialize:"true"`
-	Expiry   int64      `serialize:"true"`
+type Signature struct {
+	// Signers is a big-endian byte slice encoding which validators signed this
+	// message.
+	Signers   []byte                 `canoto:"bytes,1"`
+	Signature [bls.SignatureLen]byte `canoto:"fixed bytes,2"`
+
+	canotoData canotoData_Signature
 }
```

Comment:
Because `*warp.BitSetSignature` is implemented in avalanchego - we needed to re-implement it here to be able to serialize it differently.